torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -128,16 +132,20 @@ struct GraphFuser {
     }
   }
   bool allFloatIO(Node * node) {
+    if (!(torch::jit::isComparison(node))) { //comparisons will output byte, and that's ok
     for(auto & o : node->outputs()) {
       if(!hasFloatType(o)) {
         return false;
       }
     }
+    }
+    if (node->kind() != aten::type_as) { // as long as type_as outputs float, we are ok
     for(auto & o : node->inputs()) {
       if(!hasFloatType(o)) {
         return false;
       }
     }
+    }
```

Comment:
This feels very wrong. Having a function called `allFloatIO` return `true` even though not all inputs and outputs are float is a hack and will lead to bugs.